File: app/src/test/java/com/ivancea/MTGRules/services/RulesServiceTest.java
package com.ivancea.MTGRules.services;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.ivancea.MTGRules.model.RulesSource;

import org.junit.jupiter.api.Test;

import java.util.List;

public class RulesServiceTest {
	@Test
	public void notRepeatedRules() {
		RulesService rulesService = new RulesService(null);

		List<RulesSource> rulesSources = rulesService.getRulesSources();

		long totalDistinctRules = rulesSources.stream()
			.map(RulesSource::getDate)
			.distinct()
			.count();

		assertEquals(rulesSources.size(), totalDistinctRules, "Repeated rules");
	}
}

File: app/src/test/java/com/ivancea/MTGRules/services/RulesServiceTest.kt
package com.ivancea.MTGRules.services

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class RulesServiceTest {
    @Test
    fun notRepeatedRules() {
        val rulesSources = RulesService.rulesSources

        val totalDistinctRules = rulesSources
            .map { it.date }
            .distinct()
            .count()

        assertEquals(rulesSources.size, totalDistinctRules, "Repeated rules")
    }
}
